src/mechanics/cell/SimpleAssember.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include "mechanics/cell/CellInterface.h"
+#include "mechanics/dofs/GlobalDofVector.h"
+#include "mechanics/dofs/GlobalDofMatrixSparse.h"
+
+namespace NuTo
+{
+class SimpleAssembler
+{
+public:
+    SimpleAssembler(const NuTo::DofContainer<int>& numActiveDofs, const NuTo::DofContainer<int>& numDependentDofs)
+        : mNumActiveDofs(numActiveDofs)
+        , mNumDependentDofs(numDependentDofs)
+    {
+    }
+
+    GlobalDofVector BuildGradient(const std::vector<NuTo::CellInterface*>& cells,
+                                  const std::vector<NuTo::DofType*>& dofTypes) const
+    {
+        GlobalDofVector gradient = ProperlyResizedGlobalVector(dofTypes);
+        for (NuTo::CellInterface* cell : cells)
+        {
+            const DofVector<int> numbering = cell->DofNumbering();
+            const DofVector<double> cellGradient = cell->Gradient();
+
+            for (const DofType* dof : dofTypes)
+            {
+                const Eigen::VectorXi& numberingDof = numbering[*dof];
+                const Eigen::VectorXd& cellGradientDof = cellGradient[*dof];
+                const int numActiveDofs = mNumActiveDofs[*dof];
+                for (int i = 0; i < numberingDof.rows(); ++i)
+                {
+                    int globalDofNumber = numberingDof[i];
+                    double globalDofValue = cellGradientDof[i];
+                    if (globalDofNumber < numActiveDofs)
```

Comment:
I totally agree on the naming. I just thought: is a constrained value technically still a degree of _freedom_? We could introduce just `dofs` for all active dofs and something else for the constrained values. What do you think?

For the decision whether a dof is dependent or independent, I again like to refer to our current NuTo version. I works like a charm. This way, the assembler is not involved in the dof numbering or the constraints. This a big advantage, since it nicely separates all those tasks. 
[I am aware that this simple assembler _might_ (we never implemented it...) not be able to work in an MPI context. This might require a `class MpiAssembler` and imo is whole different problem.]